### PR TITLE
Issue #26 Adding Builder and Setter in Bound Filter

### PR DESCRIPTION
**Pull request by gagangupt16**

### CODE REVIEW

File: src/main/java/in/zapr/druid/druidry/filter/BoundFilter.java [hunk 1]
```diff
@@ -19,14 +19,18 @@
...
 
 import in.zapr.druid.druidry.SortingOrder;
-
+import lombok.Builder;
 import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
 
 @Getter
+@Setter
```
Comment (nihitavr) [inline src/main/java/in/zapr/druid/druidry/filter/BoundFilter.java]: Have we started using Setter? Previously when we worked we had decided to use Constructor or Builder for setting fields

File: src/main/java/in/zapr/druid/druidry/filter/DruidFilter.java [hunk 2]
```diff
@@ -18,9 +18,11 @@
 
 package in.zapr.druid.druidry.filter;
 
+import lombok.EqualsAndHashCode;
```
Comment (nihitavr) [inline src/main/java/in/zapr/druid/druidry/filter/DruidFilter.java]: If this works then can we use it in BoundFilter too?

Comment (GG-Zapr) [reply src/main/java/in/zapr/druid/druidry/filter/DruidFilter.java]: Will update it a separate patch. Need to fix this at lot of places.

File: src/main/java/in/zapr/druid/druidry/filter/BoundFilter.java (hunk 1 above)
Comment (GG-Zapr) [reply src/main/java/in/zapr/druid/druidry/filter/BoundFilter.java]: Setter was requested by a developer in the mail.
Have ensured that setting required as `null` would throw NullPointerException.

Comment (nihitavr) [reply src/main/java/in/zapr/druid/druidry/filter/BoundFilter.java]: If we are using Setter shouldn't we have NoArgsConstructor too?

**Pull request merged**